src/Engine/MemoryMappedFile.h

```diff
@@ -23,45 +20,33 @@
 // WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 // OTHER DEALINGS IN THE SOFTWARE.
-//
 
-#ifndef SRC_ENGINE_ASSOCIATEDPHRASES_H_
-#define SRC_ENGINE_ASSOCIATEDPHRASES_H_
+#ifndef SRC_ENGINE_MEMORYMAPPEDFILE_H_
+#define SRC_ENGINE_MEMORYMAPPEDFILE_H_
 
-#include <iostream>
-#include <map>
-#include <string>
-#include <utility>
-#include <vector>
+#include <cstddef>
 
 namespace McBopomofo {
 
-class AssociatedPhrases {
+// A wrapper for managing a memory-mapped file. Access is read-only.
+class MemoryMappedFile {
  public:
-  AssociatedPhrases();
-  ~AssociatedPhrases();
-
-  bool isLoaded();
+  ~MemoryMappedFile();
   bool open(const char* path);
   void close();
-  const std::vector<std::string> valuesForKey(const std::string& key);
-  bool hasValuesForKey(const std::string& key);
 
- protected:
-  struct Row {
-    Row(std::string_view k, std::string_view v)
-        : key(std::move(k)), value(std::move(v)) {}
-    const std::string_view key;
-    const std::string_view value;
-  };
+  [[nodiscard]] const char* data() const {
+    return static_cast<const char*>(ptr_);
+  }
 
-  std::map<std::string_view, std::vector<Row>> keyRowMap;
+  [[nodiscard]] size_t length() const { return length_; }
 
-  int fd;
-  void* data;
-  size_t length;
+ private:
+  int fd_ = -1;
```

Comment:
The naming looks confusing. When I read the code for the first time, I cannot understand what the pointer is for, Maybe just `data_`?

Reply:
Commented on what `fd_` and `ptr_` represent. PTAL.

Reply:
I misunderstood your comment. `ptr_` is now renamed to `data_`.